### PR TITLE
list_refs

### LINKED ISSUE #136: Add python support for listing all refs
We have this on the Rust side via `list_refs`

### PR DESCRIPTION
**Pull request by dcherian**
Closes #136

### CODE REVIEW

File: icechunk-python/python/icechunk/_icechunk_python.pyi [hunk 1]
```diff
@@ -3,6 +3,12 @@ import datetime
 from collections.abc import AsyncGenerator
 from typing import Any
 
+class Ref:
```
Comment (dcherian) [inline icechunk-python/python/icechunk/_icechunk_python.pyi:6]: Is this really the model we want to return to users?

Alternatives could be:
1. Arraylake style Pydantic models: `BranchName(str)` and `TagName(str)`
2. or, 
```
class RefType(Enum):
    BRANCH = auto()
    TAG = auto()

class Ref:
    name: str
    kind: RefType
```

File: icechunk-python/src/lib.rs [hunk 2]
```diff
@@ -40,6 +40,38 @@ struct PyIcechunkStore {
...
+    fn from(reference: Ref) -> Self {
+        match reference {
+            Ref::Tag(name) => Self::Tag { name },
+            Ref::Branch(name) => Self::Branch { name },
+        }
+    }
+}
+
+#[pymethods]
+impl PyRef {
```
Comment (paraseba) [inline icechunk-python/src/lib.rs:60]: beautiful

File: icechunk-python/src/lib.rs [hunk 3]
```diff
@@ -432,6 +464,25 @@ impl PyIcechunkStore {
         Ok(snapshot_id.to_string())
     }
 
+    fn list_refs<'py>(
```
Comment (paraseba) [inline icechunk-python/src/lib.rs:467]: can we also provide `async_list_refs`?

Comment (paraseba) [reply icechunk-python/src/lib.rs:467]: can we add a python test that exercises this new function?

Comment (dcherian) [reply icechunk-python/src/lib.rs:467]: Yes I can do both. I'd like some input on https://github.com/earth-mover/icechunk/pull/318#discussion_r1813703484 first before I tackle those.

File: icechunk-python/python/icechunk/_icechunk_python.pyi (hunk 1 above)
Comment (paraseba) [reply icechunk-python/python/icechunk/_icechunk_python.pyi:6]: Oh sorry, forgot to comment. i like your approach best, but I'd like to see how easy/hard is to pattern match the result in python. Probably tests will show me that.

Other option would be to have separate functions, `list_tags` and `list_branches`, or maybe the three of them.

Comment (mpiannucci) [reply icechunk-python/python/icechunk/_icechunk_python.pyi:6]: I think that tags and branches can be in separate functions and that simplifies a lot from the Python user perspective

**Comment by dcherian** [comment]
@mpiannucci has implemented this

**Pull request closed without merging**